airflow/operators/mysql_to_s3_operator.py

```diff
@@ -0,0 +1,122 @@
+    :type mysql_conn_id: str
+     :param aws_conn_id: reference to a specific S3 connection
+    :type aws_conn_id: str
+    :param verify: Whether or not to verify SSL certificates for S3 connection.
+        By default SSL certificates are verified.
+        You can provide the following values:
+
+        - False: do not validate SSL certificates. SSL will still be used
+                 (unless use_ssl is False), but SSL certificates will not be
+                 verified.
+        - path/to/cert/bundle.pem: A filename of the CA cert bundle to uses.
+                 You can specify this argument if you want to use a different
+                 CA cert bundle than the one used by botocore.
+    :type verify: bool or str
+    :param header: whether to include header or not into the S3 file
+    :type header: bool
+    :param index: whether to include index or not into the S3 file
+    :type index: bool
+    """
+
+    @apply_defaults
+    def __init__(
+            self,
+            query: str,
+            s3_bucket: str,
+            s3_key: str,
+            mysql_conn_id: str = 'mysql_default',
+            aws_conn_id: str = 'aws_default',
+            verify: Optional[Union[bool, str]] = None,
+            header: Optional[bool] = False,
+            index: Optional[bool] = False,
+            *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.query = query
+        self.s3_bucket = s3_bucket
+        self.s3_key = s3_key
+        self.mysql_conn_id = mysql_conn_id
+        self.aws_conn_id = aws_conn_id
+        self.verify = verify
+        self.header = header
+        self.index = index
+
+    def fill_na_with_none(self, series):
+        """
+        Replace NaN values with None
+        """
+        return np.where(series.isnull(), None, series)
+
+    def fix_int_dtypes(self, df):
+        """
+        Mutate DataFrame to set dtypes for int columns containing NaN values."
+        """
+        for col in df:
+            if "float" in df[col].dtype.name and df[col].hasnans:
+                # inspect values to determine if dtype of non-null values is int or float
+                notna_series = df[col].dropna().values
+                if np.isclose(notna_series, notna_series.astype(int)).all():
+                    # set to dtype that retains integers and supports NaNs
+                    df[col] = self.fill_na_with_none(df[col]).astype(pd.Int64Dtype)
+
+    def execute(self, context):
+        mysql_hook = MySqlHook(mysql_conn_id=self.mysql_conn_id)
+        s3_conn = S3Hook(aws_conn_id=self.aws_conn_id, verify=self.verify)
+        data_df = mysql_hook.get_pandas_df(self.query)
+        self.log.info("Data from MySQL obtained")
+
+        self.fix_int_dtypes(data_df)
+        with tempfile.NamedTemporaryFile(suffix='.csv') as tmp_csv:
+            data_df.to_file(tmp_csv)
+            self.log.info("Temporary csv created")
+            s3_conn.load_file(tmp_csv,
```

Comment:
```suggestion
            s3_conn.load_file(tmp_csv.name,
```
I think we should point to name